src/wp-includes/block-template-utils.php

```diff
@@ -1444,36 +1477,40 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
  * @since 6.5.0
  * @access private
  *
- * @param stdClass        $post    An object representing a template or template part
+ * @param stdClass        $changes An object representing a template or template part
  *                                 prepared for inserting or updating the database.
  * @param WP_REST_Request $request Request object.
  * @return stdClass The updated object representing a template or template part.
  */
-function inject_ignored_hooked_blocks_metadata_attributes( $post, $request ) {
-	$filter_name = current_filter();
-	if ( ! str_starts_with( $filter_name, 'rest_pre_insert_' ) ) {
-		return $post;
-	}
-	$post_type = str_replace( 'rest_pre_insert_', '', $filter_name );
-
+function inject_ignored_hooked_blocks_metadata_attributes( $changes, $request ) {
 	$hooked_blocks = get_hooked_blocks();
 	if ( empty( $hooked_blocks ) && ! has_filter( 'hooked_block_types' ) ) {
-		return $post;
+		return $changes;
+	}
+
+	$terms = $changes->tax_input;
+
+	if ( ! empty( $changes->ID ) ) {
+		$type_terms        = get_the_terms( $changes->ID, 'wp_theme' );
+		$terms['wp_theme'] = ! is_wp_error( $type_terms ) && ! empty( $type_terms ) ? $type_terms[0]->name : null;
+
+		// Apply changes to the existing post object.
+		$post = get_post( $changes->ID );
+		$post = (object) array_merge( (array) $post, (array) $changes );
+	} else {
+		if ( empty( $changes->post_name ) ) {
+			$changes->post_name = $request['slug'];
+		}
+		$post = $changes;
 	}
 
-	// At this point, the post has already been created.
-	// We need to build the corresponding `WP_Block_Template` object as context argument for the visitor.
-	// To that end, we need to suppress hooked blocks from getting inserted into the template.
-	add_filter( 'hooked_block_types', '__return_empty_array', 99999, 0 );
-	$template = $request['id'] ? get_block_template( $request['id'], $post_type ) : null;
-	remove_filter( 'hooked_block_types', '__return_empty_array', 99999 );
+	$template = _build_block_template_object_from_wp_post_object( new WP_Post( $post ), $terms, $changes->meta_input );
```

Comment:
Same as above. Since `_build_block_template_object_from_wp_post_object` can return a `WP_Error` shall we check that here before passing `$template` in as `$context`?

Reply:
https://github.com/WordPress/wordpress-develop/pull/6278/commits/3645487dccd030c2fb35a4764cffdda55bd7a044